### ISSUE TITLE
ELECTRA: Pre-training Text Encoders as Discriminators Rather Than Generators

### ISSUE THREAD

**Issue by hs-sugi** [One-Day-One-Paper/One-Day-One-Paper#187]
- [Submitted on 23 Mar 2020]
- Kevin Clark, Minh-Thang Luong, Quoc V. Le, Christopher D. Manning

### Abstract
- BERT 등 기존 language model들은 입력을 마스크 토큰으로 치환, 마지막에는 치환 전의 원본 토큰으로 복원함 => masked language modeling 태스크 pre-training
- 계산량이 많다는 단점을 극복하고자 효율을 향상시키기 위해 Replaced Token Detection (RTD) pre--training 태스크 제안
- 결과적으로 BERT의 성능 능가, small 모델의 경우 GPT 능가, large모델의 경우 RoBERTa나 XLNet 대비 1/4의 계산량으로 비슷한 성능 보임
- METHOD
<img width="592" alt="image" src="https://user-images.githubusercontent.com/40443049/169701314-87ff3489-acc8-432a-aaa6-b2ba0a414819.png">
- RTD 태스크를 통해 학습하기 위해서 generator G 와 discriminator D, 두개의 네트워크 필요
- 두 네트워크는 공통적으로 Transformer 인코더 구조 
- 토큰시퀀스 x = [ x1, x2,..., xn] 을 입력 받아 문맥 정보 반영한 벡터 시퀀스 h(x) = [h1, h2, ..., hn]로 매핑
- Generator
  - BERT의 MLM과 동일
  - loss functions
<img width="767" alt="image" src="https://user-images.githubusercontent.com/40443049/169701496-2b65b8fd-d259-4f62-b4db-fa94be5bc00f.png">
- Discriminator 
  - 입력 토큰 시퀀스에 대해서 각 토큰이 original인지 replaced인지 이진 분류로 학습
  - loss functions
<img width="767" alt="image" src="https://user-images.githubusercontent.com/40443049/169701565-f6414f1c-365a-4a7e-bf82-cebd0725e92b.png">
- 대랸의 코퍼스에 대해서 generator loss와 discriminator loss의 합을 최소화하도록 학습
- 샘플링 과정으로 discriminator loss는 generator로 역전파 되지 않음
-  pre-training을 마친 뒤에 generator를 버리고 discriminator만 취해서 downstream task으로 fine-tuning 진행
<img width="783" alt="image" src="https://user-images.githubusercontent.com/40443049/169701626-fe1616fb-d65a-414f-9779-f210c7c823d4.png">
<img width="783" alt="image" src="https://user-images.githubusercontent.com/40443049/169701632-7163e66b-c548-4740-9533-905aec92b5d9.png">
<img width="783" alt="image" src="https://user-images.githubusercontent.com/40443049/169701645-09b9c0b5-98b7-4e0e-a450-b9e0aeab5aa7.png">
